src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2879,7 +2843,7 @@ public override bool ContainsSymbolsWithName(Func<string, bool> predicate, Symbo
                 throw new ArgumentException(CSharpResources.NoNoneSearchCriteria, nameof(filter));
             }
 
-            return _declarationTable.ContainsName(predicate, filter, cancellationToken);
+            return _syntaxAndDeclarations.GetLazyState().DeclarationTable.ContainsName(predicate, filter, cancellationToken);
```

Comment:
Consider using Declarations instead of _syntaxAndDeclarations.GetLazyState().DeclarationTable, here and elsewhere.